pkg/cliplugins/bind/plugin/bind_compute.go

```diff
@@ -0,0 +1,295 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/kcp-dev/logicalcluster/v2"
+	"github.com/martinlindhe/base36"
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	schedulingv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/scheduling/v1alpha1"
+	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	"github.com/kcp-dev/kcp/pkg/cliplugins/base"
+	"github.com/kcp-dev/kcp/pkg/cliplugins/helpers"
+)
+
+type BindWorkloadOptions struct {
+	*base.Options
+
+	// APIExports is a list of APIExport to use in the workspace.
+	APIExports []string
+
+	// Namespace selector is a label selector to select namespace for the workload.
+	NamespaceSelector string
+
+	// LocationSelectors is a list of label selectors to select locations in the compute workspace.
+	LocationSelectors []string
+
+	// ComputeWorkspace is the workspace for synctarget
+	ComputeWorkspace logicalcluster.Name
+}
+
+func NewBindWorkloadOptions(streams genericclioptions.IOStreams) *BindWorkloadOptions {
+	return &BindWorkloadOptions{
+		Options: base.NewOptions(streams),
+	}
+}
+
+// BindFlags binds fields SyncOptions as command line flags to cmd's flagset.
+func (o *BindWorkloadOptions) BindFlags(cmd *cobra.Command) {
+	o.Options.BindFlags(cmd)
+
+	cmd.Flags().StringSliceVar(&o.APIExports, "apiexports", o.APIExports,
+		"APIExport to bind to this workspace for workload, each APIExoport should be in the format of <absolute_ref_to_workspace>:<apiexport>")
+	cmd.Flags().StringVar(&o.NamespaceSelector, "namespace-selector", o.NamespaceSelector, "Label select to select namespaces to create workload.")
+	cmd.Flags().StringSliceVar(&o.LocationSelectors, "location-selectors", o.LocationSelectors,
+		"A list of label selectors to select locations in the compute workspace to sync workload.")
+}
+
+// Complete ensures all dynamically populated fields are initialized.
+func (o *BindWorkloadOptions) Complete(args []string) error {
+	if err := o.Options.Complete(); err != nil {
+		return err
+	}
+
+	if len(args[0]) != 1 {
+		return fmt.Errorf("a compute workspace should be specified")
+	}
+
+	o.ComputeWorkspace = logicalcluster.New(args[0])
```

Comment:
There's a helper for `NewValid()` - we should use that & indicate to a user when the input is invalid